common/app/common/dfp/PageskinAdAgent.scala

```diff
@@ -1,42 +1,51 @@
 package common.dfp
 
 import common.Edition
-import conf.Configuration.commercial._
+import com.gu.commercial.display.AdTargetParam.toMap
+import com.gu.commercial.display.MultipleValues
+import model.MetaData
 
 trait PageskinAdAgent {
 
   protected val environmentIsProd: Boolean
 
   protected def pageSkinSponsorships: Seq[PageSkinSponsorship]
 
-  private def findSponsorships(adUnitWithoutRoot: String, edition: Edition): Seq[PageSkinSponsorship] = {
+  // There are two forms of pageskins:
+  // - pageskins that target through ad unit (for pressed fronts)
+  // - pageskins that target through a keyword (for index page fronts)
```

Comment:
Can you codify this with functions rather than writing in a comment? By this I mean take the bodies of the if and else clause and create functions from them, invoking them in each of the instances you mention. I.e pageskins that target through ad unit and those that target through keyword.

Reply:
the code is pretty clear due to the `if` block, and pretty small. it's just the business logic won't ever come through clearly (the why). It would have to be something like `adUnitTargetedPageskinsForPressedFronts`, I think gains some ambiguity due to the lack of grammar.